flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -49,7 +49,11 @@
  *
  * <p>The size of this pool can be dynamically changed at runtime ({@link #setNumBuffers(int)}. It
  * will then lazily return the required number of buffers to the {@link NetworkBufferPool} to match
- * its new size.
+ * its new size. New buffers can be requested only when {@code numberOfRequestedMemorySegments +
+ * numberOfRequestedOverdraftMemorySegments < currentPoolSize + maxOverdraftBuffersPerGate}. In
```

Comment:
This PR just ensure that new buffers cannot be requested when `the total number of requested buffers` exceeds `poolSize + maxOverdraftBuffersPerGate`, and cannot ensure that `the total number of requested buffers does not exceed poolSize + maxOverdraftBuffersPerGate`, right?

If so, the commit message and JIRA title should be changed.

Reply:
Yes, good catch! I was stupid when typing the commit message. 🤣 

In fact, maybe we can never guarantee that `the total number of requested buffers does not exceed poolSize + maxOverdraftBuffersPerGate`, because we will not force buffers to return when the pool size changes, right?